server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -669,6 +671,18 @@ protected Node(
                 clusterService.getClusterSettings(),
                 threadPool::relativeTimeInMillis
             );
+            final RemoteClusterStateService remoteClusterStateService;
+            if (RemoteClusterStateService.REMOTE_CLUSTER_STATE_ENABLED_SETTING.get(settings) == true) {
+                remoteClusterStateService = new RemoteClusterStateService(
+                    nodeEnvironment.nodeId(),
+                    repositoriesServiceReference::get,
+                    settings,
+                    clusterService.getClusterSettings(),
+                    threadPool::preciseRelativeTimeInNanos
+                );
+            } else {
+                remoteClusterStateService = null;
+            }
```

Comment:
[minor] dont need it explicitly, just initialize with null.

Reply:
Since the variable is final, we cannot initialize it with null. Final variable is need later while injecting.